sdk/core/azure-core/src/main/java/com/azure/core/util/RequestContent.java

```diff
@@ -0,0 +1,288 @@
+import com.azure.core.implementation.util.SerializableContent;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.core.util.serializer.JsonSerializerProviders;
+import com.azure.core.util.serializer.ObjectSerializer;
+import reactor.core.publisher.Flux;
+
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.util.Objects;
+
+/**
+ * Represents the content sent as part of a request.
+ */
+public abstract class RequestContent {
+    /**
+     * Converts the {@link RequestContent} into a {@code Flux<ByteBuffer>} for use in reactive streams.
+     *
+     * @return The {@link RequestContent} as a {@code Flux<ByteBuffer>}.
+     */
+    public abstract Flux<ByteBuffer> asFluxByteBuffer();
+
+    /**
+     * Gets the length of the {@link RequestContent} if it is able to be calculated.
+     * <p>
+     * If the content length isn't able to be calculated null will be returned.
+     *
+     * @return The length of the {@link RequestContent} if it is able to be calculated, otherwise null.
+     */
+    public abstract Long getLength();
+
+    /**
+     * Creates a {@link RequestContent} that uses {@code byte[]} as its data.
+     *
+     * @param bytes The bytes that will be the {@link RequestContent} data.
+     * @return A new {@link RequestContent}.
+     * @throws NullPointerException If {@code bytes} is null.
+     */
+    public static RequestContent fromBytes(byte[] bytes) {
+        Objects.requireNonNull(bytes, "'bytes' cannot be null.");
+        return fromBytes(bytes, 0, bytes.length);
+    }
+
+    /**
+     * Creates a {@link RequestContent} that uses {@code byte[]} as its data.
+     *
+     * @param bytes The bytes that will be the {@link RequestContent} data.
+     * @param offset Offset in the bytes where the data will begin.
+     * @param length Length of the data.
+     * @return A new {@link RequestContent}.
+     * @throws NullPointerException If {@code bytes} is null.
+     * @throws IllegalArgumentException If {@code offset} or {@code length} are negative or {@code offset} plus {@code
+     * length} is greater than {@code bytes.length}.
+     */
+    public static RequestContent fromBytes(byte[] bytes, int offset, int length) {
+        Objects.requireNonNull(bytes, "'bytes' cannot be null.");
+        if (offset < 0) {
+            throw new ClientLogger(RequestContent.class).logExceptionAsError(new IllegalArgumentException(
+                "'offset' cannot be negative."));
+        }
+        if (length < 0) {
+            throw new ClientLogger(RequestContent.class).logExceptionAsError(new IllegalArgumentException(
+                "'length' cannot be negative."));
+        }
+        if (offset + length > bytes.length) {
+            throw new ClientLogger(RequestContent.class).logExceptionAsError(new IllegalArgumentException(
```

Comment:
Consider creating a static final logger as it's used in multiple places.

Reply:
Yes, I should do that, before it wasn't done as `RequestContent` was an interface, but now that it is an abstract class that isn't an issue.